ksqldb-execution/src/main/java/io/confluent/ksql/execution/interpreter/terms/ColumnReferenceTerm.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.interpreter.terms;
+
+import io.confluent.ksql.execution.interpreter.TermEvaluationContext;
+import io.confluent.ksql.execution.interpreter.terms.BasicTerms.BooleanTerm;
+import io.confluent.ksql.schema.ksql.types.SqlBaseType;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+
+public class ColumnReferenceTerm implements Term {
+
+  private final int rowIndex;
+  private final SqlType sqlType;
+
+  public ColumnReferenceTerm(
+      final int rowIndex,
+      final SqlType sqlType
+  ) {
+    this.rowIndex = rowIndex;
+    this.sqlType = sqlType;
+  }
+
+  @Override
+  public Object getValue(final TermEvaluationContext context) {
+    return context.getRow().get(rowIndex);
+  }
+
+  @Override
+  public SqlType getSqlType() {
+    return sqlType;
+  }
+
+  public static ColumnReferenceTerm create(
+      final int rowIndex,
+      final SqlType sqlType
+  ) {
+    if (sqlType.baseType() == SqlBaseType.BOOLEAN) {
+      return new BooleanColumnReferenceTerm(rowIndex, sqlType);
+    }
+    return new ColumnReferenceTerm(rowIndex, sqlType);
+  }
+
+  public static class BooleanColumnReferenceTerm extends ColumnReferenceTerm
```

Comment:
why do we have a special one for booleans? (probably worth some javadoc)

Reply:
I ended up getting rid of all of the interfaces and special casing for BooleanTerms.  It's a good idea and I think it'll make sense to do in the future, but it introduced edge cases in the current code.  I realized that the `CastTerm` didn't return a `BooleanTerm` even when the type was a boolean result in a casting exception.  I would have had to special case the Cast code, which just adds additional complexity to it and future changes which would have to ensure to special case Boolean.  It also begs the question of why Boolean and not others.  When I have a more complete picture in the future that doesn't require special casing, I'll tackle it then.